pricegraph/bench/pricegraph.rs

```diff
@@ -14,29 +15,35 @@ pub fn read(c: &mut Criterion) {
 pub fn transitive_orderbook(c: &mut Criterion) {
     let pricegraph = read_default_pricegraph();
     let dai_weth = Market { base: 7, quote: 1 };
+    let hops = &[None, Some(1), Some(2), Some(5), Some(10), Some(30)];
 
-    c.bench_with_input(
-        BenchmarkId::new("Pricegraph::transitive_orderbook", *DEFAULT_BATCH_ID),
-        &(&pricegraph, dai_weth),
-        |b, &(pricegraph, dai_weth)| {
-            b.iter(|| pricegraph.transitive_orderbook(dai_weth, None, None))
-        },
-    );
+    let mut group = c.benchmark_group("Pricegraph::transitive_orderbook");
+    for hops in hops {
+        group.bench_with_input(
+            BenchmarkId::from_parameter(format!("hops:{:?}", hops)),
+            &(&pricegraph, dai_weth, hops),
+            |b, &(pricegraph, dai_weth, hops)| {
+                b.iter(|| pricegraph.transitive_orderbook(dai_weth, *hops, None))
+            },
+        );
+    }
+    group.finish();
 }
 
 pub fn estimate_limit_price(c: &mut Criterion) {
     let pricegraph = read_default_pricegraph();
-    let dai_weth = TokenPair { buy: 7, sell: 1 }.into_unbounded_range();
+    let dai_weth = TokenPair { buy: 7, sell: 1 };
     let eth = 1e18;
     let volumes = &[0.1 * eth, eth, 10.0 * eth, 100.0 * eth, 1000.0 * eth];
+    let hops = &[None, Some(1), Some(2), Some(5), Some(10), Some(30)];
 
     let mut group = c.benchmark_group("Pricegraph::estimate_limit_price");
-    for volume in volumes {
+    for (volume, hops) in volumes.iter().cartesian_product(hops) {
```

Comment:
Fancy! I like this as it reduces nesting, but I'm surprised you didn't just go for nested for loops :stuck_out_tongue: